chips/sam4l/src/ast.rs

```diff
@@ -28,30 +28,138 @@ const ALARM0_SYNC_TICS: u32 = 8;
+    wer: ReadWrite<u32, Event::Register>,
     // 0x20
-    ar0: VolatileCell<u32>,
-    ar1: VolatileCell<u32>,
+    ar0: ReadWrite<u32, Value::Register>,
+    ar1: ReadWrite<u32, Value::Register>,
     _reserved0: [u32; 2],
-    pir0: VolatileCell<u32>,
-    pir1: VolatileCell<u32>,
+    pir0: ReadWrite<u32, PeriodicInterval::Register>,
+    pir1: ReadWrite<u32, PeriodicInterval::Register>,
     _reserved1: [u32; 2],
     // 0x40
-    clock: VolatileCell<u32>,
-    dtr: VolatileCell<u32>,
-    eve: VolatileCell<u32>,
-    evd: VolatileCell<u32>,
-    evm: VolatileCell<u32>,
-    calv: VolatileCell<u32>, // we leave out parameter and version
+    clock: ReadWrite<u32, ClockControl::Register>,
+    dtr: ReadWrite<u32, DigitalTuner::Register>,
+    eve: WriteOnly<u32, Event::Register>,
+    evd: WriteOnly<u32, Event::Register>,
+    evm: ReadOnly<u32, Event::Register>,
+    calv: ReadWrite<u32, Calendar::Register>, // we leave out parameter and version
 }
 
+register_bitfields![u32,
+    Control [
+        /// Prescalar Select
+        PSEL OFFSET(16) NUMBITS(5) [],
+        /// Clear on Alarm 1
+        CA1  OFFSET(9) NUMBITS(1) [
+            NoClearCounter = 0,
+            ClearCounter = 1
+        ],
+        /// Clear on Alarm 0
+        CA0  OFFSET(8) NUMBITS(1) [
+            NoClearCounter = 0,
+            ClearCounter = 1
+        ],
+        /// Calendar Mode
+        CAL  OFFSET(2) NUMBITS(1) [
+            CounterMode = 0,
+            CalendarMode = 1
+        ],
+        /// Prescalar Clear
+        PCLR OFFSET(1) NUMBITS(1) [],
+        /// Enable
+        EN   OFFSET(0) NUMBITS(1) [
+            Disable = 0,
+            Enable = 1
+        ]
+    ],
+
+    Value [
+        VALUE OFFSET(0) NUMBITS(32) []
+    ],
+
+    Status [
+        /// Clock Ready
+        CLKRDY 29,
+        /// Clock Busy
+        CLKBUSY 28,
+        /// AST Ready
+        READY 25,
+        /// AST Busy
+        BUSY 24,
+        /// Periodic 0
+        PER0 16,
+        /// Alarm 0
+        ALARM0 8,
+        /// Overflow
+        OVF 0
+    ],
+
+    Interrupt [
+        /// Clock Ready
+        CLKRDY 29,
+        /// AST Ready
+        READY 25,
+        /// Periodic 0
+        PER0 16,
+        /// Alarm 0
+        ALARM0 8,
+        /// Overflow
+        OVF 0
+    ],
+
+    Event [
+        /// Periodic 0
+        PER0 16,
+        /// Alarm 0
+        ALARM0 8,
+        /// Overflow
+        OVF 0
+    ],
+
+    PeriodicInterval [
+        /// Interval Select
+        INSEL OFFSET(0) NUMBITS(5) []
+    ],
+
+    ClockControl [
+        /// Clock Source Selection
+        CSSEL OFFSET(8) NUMBITS(3) [
+            RCSYS = 0,
+            OSC32 = 1,
+            APBClock = 2,
+            GCLK = 3,
+            Clk1k = 4
+        ],
+        /// Clock Enable
+        CEN   OFFSET(0) NUMBITS(1) [
+            Disable = 0,
+            Enable = 1
+        ]
+    ],
+
+    DigitalTuner [
+        VALUE OFFSET(8) NUMBITS(8) [],
```

Comment:
everything above got comments, why quit here? :D

Reply:
Pretty much only because the datasheet gave up